docs/more/development-tips.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Tips for developing Yew applications
+---
+
+:::important contribute
+This document only contains information for adding supporting in IntelliJ IDEA and VSCode.
+Feel free to contribute to add instructions for your editor of choice. 
+:::
+
+## Add a template for creating components
+
+### IntelliJ IDEA
+
+1. Navigate to File | Settings | Editor | Live Templates.  
+2. Select Rust and click on + icon to add a new Live Template.
+3. Give it a name and description of your preference.
+4. Paste the following snippet in Template Text section:
+
+```rust
+use yew::prelude::*;
+
+struct $NAME$ {
+    link: ComponentLink<Self>
+}
+
+enum Msg {
+}
+
+impl Component for $NAME$ {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(_props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        Self { 
+            link 
+        }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {}
+    }
+
+    fn change(&mut self, _props: Self::Properties) -> ShouldRender {
+        false
+    }
+
+    fn view(&self) -> Html {
+        html! {
+            $HTML$
+        }
+    }
+}
+```
+
+### VSCode
+
+1. Navigate to File | Preferences | User Snippets.
+2. Select Rust as the language.
+3. Add the following snippet in the snippet JSON file:
+```json
+{
+	"Create new Yew component": {
+		"prefix": "YOUR PREFIX OF CHOICE",
+		"body": [
+			"use yew::prelude::*;",
+			"",
+			"pub struct ${1} {",
+			"    link: ComponentLink<Self>,",
+			"}",
+			"",
+			"pub enum Msg {",
+			"}",
+			"",
+			"impl Component for ${1} {",
+			"    type Message = Msg;",
+			"    type Properties = ();",
+			"",
+			"    fn create(_props: Self::Properties, link: ComponentLink<Self>) -> Self {",
+			"        Self {",
+			"            link,",
+			"        }",
+			"    }",
+			"",
+			"    fn update(&mut self, msg: Self::Message) -> ShouldRender {",
+			"        match msg {}",
+			"    }",
+			"",
+			"    fn change(&mut self, _props: Self::Properties) -> ShouldRender {",
+			"        false",
+			"    }",
+			"",
+			"    fn view(&self) -> Html {",
+			"        html! {",
+			"            ${0}",
+			"        }",
+			"    }",
+			"}"
+		],
+		"description": "Create new Yew component"
+	}
+}
+
+```
+
+## Enable HTML intellisense for `html!` 
+
+### IntelliJ IDEA
+
+There is currently no way to extend the support for proc macros in `intellij-rust`. See [intellij-rust/intellij-rust#6367](https://github.com/intellij-rust/intellij-rust/issues/6367) 
```

Comment:
Isn't this just the case for `proc-macro-hack` – otherwise I think they work fine. As we've already removed proc-macro-hack things should "just work" in the next release no?

Reply:
No, there's still no support for everything that goes *inside* the macro. Removing `proc-macro-hack` only helped in the return type.

Reply:
There is "support," just more from rust-analzyer's perspective than ours :)

I guess it's good to clarify this in the docs.

Reply:
I'm not sure what is there to clarify. What else should go there?